services/internal-lb-service/README.md

```diff
@@ -0,0 +1,139 @@
+# Internal Load Balancer Service
+
+Internal Load Balancing on GKE deploys an internal TCP/UDP load balancer for private L4 load balancing. This example deploys an application on GKE and exposes the application with a private load balanced IP address. See the [internal-lb-service.yaml](internal-lb-service.yaml) manifest for the full deployment spec.
+
+### Use-cases
+
+- Private exposure of a GKE internal application
+- TCP or UDP Load Balancing for private Services
+
+### Relevant documentation
+
+- [Services of type LoadBalancer](https://cloud.google.com/kubernetes-engine/docs/concepts/service#services_of_type_loadbalancer).  **Note:** an annotation is required to ensure an internal LB is created and not an external LB.
+- [Creating an internal TCP load balancer Service](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing#create)
+- [Optional alternative subnet for GKE internal load balancers](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing#lb_subnet)
+
+### Versions
+
+- All supported GKE versions
+
+
+![internal LoadBalancer service](../../images/internal-lb-service.png)
+
+### Networking Manifests
+
+In this example, a Service matches TCP traffic destined to port 80 and load balances across pods in the `foo` Deployment on TCP port 8080.  Key aspects of this manifest include:
+
+- The `type: LoadBalancer` specification and the metadata annotation `cloud.google.com/load-balancer-type: "Internal"` that prompt GKE to create an internal TCP/UDP Load Balancer for this Service
+- The `spec.ports.port` to define the port that the internal load balancer will listen on, whereas the `spec.ports.targetPort` defines the port that the pods in the Deployment will listen on.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: foo
+  annotations:
+    cloud.google.com/load-balancer-type: "Internal"
+  labels:
+    app: foo
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+    protocol: TCP
+  selector:
+    app: foo
+  type: LoadBalancer
+```
+
+
+### Try it out
+
+1. Download this repo and navigate to this folder.
+
+```sh
+$ git clone https://github.com/GoogleCloudPlatform/gke-networking-recipes.git
+Cloning into 'gke-networking-recipes'...
+
+$ cd gke-networking-recipes/internal-lb-service
+```
+
+2. Deploy the Deployment and Service resources in the [internal-lb-service.yaml](internal-lb-service.yaml) manifest.
+
+```sh
+$ kubectl apply -f internal-lb-service.yaml
+service/foo created
+deployment.apps/foo created
+```
+
+3. It may take up to a minute for the pods to deploy and up to a few minutes for the internal IP address of the Service resource to be ready.  Ensure [Google Cloud probers](https://cloud.google.com/load-balancing/docs/health-check-concepts#ip-ranges) are allowed in the VPC Firewall rules so health-checks can be used. Validate the progress and make sure that no errors are surfaced in the resource events.
```

Comment:
It may take up to a minute for the pods to deploy and up to a few minutes for the internal IP address of the Service resource to be ready. Validate the progress and make sure that no errors are surfaced in the resource events. [Google Cloud health checks](https://cloud.google.com/load-balancing/docs/health-check-concepts#ip-ranges) are created within your VPC by the GKE Service controller so that health checks are allowed to reach your cluster. If the load balancer health checks appear to not be passing, check that the correct VPC firewall rules are installed.

Reply:
Made suggested change and committed. Thanks for your review!